zilliqa/src/node_launcher.rs

```diff
@@ -123,12 +121,8 @@ impl NodeLauncher {
                 () = &mut sleep => {
                     trace!("timeout elapsed");
 
-                    if !joined {
-                        self.outbound_message_sender.send((None, self.config.eth_chain_id, Message::External(ExternalMessage::JoinCommittee(self.secret_key.node_public_key())))).unwrap();
-                        joined = true;
-                    } else {
-                        self.node.lock().unwrap().handle_timeout().unwrap();
-                    }
+                    self.node.lock().unwrap().handle_timeout().unwrap();
+                    self.outbound_message_sender.send((None, self.config.eth_chain_id, Message::External(ExternalMessage::Hello(self.secret_key.node_public_key())))).unwrap();
```

Comment:
Does this mean we send both a `NewView` (if we're already part of the committee) and a `Hello` on timeout?

Reply:
Yes, although I will remove the `Hello` I think.